lib/buffer.js

```diff
@@ -139,27 +140,32 @@ function alignPool() {
   }
 }
 
-var bufferWarn = true;
+var bufferWarningAlreadyEmitted = false;
+var nodeModulesCheckCounter = 0;
 const bufferWarning = 'The Buffer() and new Buffer() constructors are not ' +
                       'recommended for use due to security and usability ' +
                       'concerns. Please use the new Buffer.alloc(), ' +
                       'Buffer.allocUnsafe(), or Buffer.from() construction ' +
                       'methods instead.';
 
 function showFlaggedDeprecation() {
-  if (bufferWarn) {
-    // This is a *pending* deprecation warning. It is not emitted by
-    // default unless the --pending-deprecation command-line flag is
-    // used or the NODE_PENDING_DEPRECATION=1 env var is set.
-    process.emitWarning(bufferWarning, 'DeprecationWarning', 'DEP0005');
-    bufferWarn = false;
+  if (bufferWarningAlreadyEmitted ||
+      nodeModulesCheckCounter > 10000 ||
```

Comment:
why not just `++nodeModulesCheckCounter > 10000` and get rid of the other one

Reply:
@devsnek yeah, I guess that makes just as much sense :)